scripts/addr2line.py

```diff
@@ -26,7 +26,18 @@
 from enum import Enum
 
 class Addr2Line:
-    def __init__(self, binary, concise=False):
+
+    # Matcher for a line that appears at the end a single decoded
+    # address, which we force by adding a dummy 0x0 address. The
+    # pattern varies between binutils addr2line and llvm-addr2line
+    # so we match both.
+    dummy_pattern = re.compile(
+        r"(.*0x0000000000000000: \?\? \?\?:0\n)" # addr2line pattern
+        r"|"
+        r"(.*0x0: \?\? at \?\?:0\n)"  # llvm-addr2line pattern
+        )
```

Comment:
The main complication is the slightly differnet "end of backtrace" beahvior for llvm-addr2line: the 0x0 frame we use to find it looks a bit different. Use a regex to accommodate the differences.